_posts/2021-05-21-js-waku.md

```diff
@@ -0,0 +1,114 @@
+The nim-waku project doubles as a library, that can be used to add Waku v2 support to native applications.
+
+<!-- present the gap js-waku is trying to fill -->
+
+We believe that dApps and wallets can benefit from the Waku network in several ways.
+For some dApps, it may make sense to enable peer-to-peer communications.
+For others, machine-to-machine communications could be a great asset.
+For example, in the case of a DAO with a governance token,
+Waku could be used for [gas-less polling](https://twitter.com/ethstatus/status/1293963402705018881?s=20).
+Enabling the DAO to notify their users of a new poll, and users to vote without interacting with the blockchain.
+
+<!-- explain how js-waku is filling the gap -->
+
+For this to happen, we need to give developers a library which is easy integrate in their dApps:
+[JS-Waku](https://github.com/status-im/js-waku).
+
+<!-- explain what we have achieved so far: more on the journey? -->
+
+JS-Waku is a JavaScript library that allows your dApp, wallet or other web app to interact with the Waku v2 network.
+It is available right now on [npm](https://www.npmjs.com/package/js-waku): `npm install js-waku`.
+
+As it is written in TypeScript, types are included in the npm package to allow easy integration with TypeScript, ClojureScript and other typed languages that compile to JavaScript.
+
+Key Waku v2 protocols are already available:
+[message](https://rfc.vac.dev/spec/14/), [store](https://rfc.vac.dev/spec/13/), [relay](https://rfc.vac.dev/spec/11/) and [light push](https://rfc.vac.dev/spec/19/),
+enabling your dApp to:
+
+- Send and receive near-instant messages on the Waku network (relay),
+- Query nodes for messages that may have been missed, e.g. due to poor cellular network (store),
+- Send messages with confirmations (light push).
+
+We focused the past month on developing a [ReactJS Chat App](https://status-im.github.io/js-waku/).
+The aim was to create enough building blocks in JS-Waku to enable this showcase web app.
+Most of the focus was on getting familiar with both the Waku v2 [RFCs](https://rfc.vac.dev/)
+and the [js-libp2p](https://github.com/libp2p/js-libp2p) library that we heavily rely on.
+Soon, we will [use it daily for dogfooding](https://github.com/status-im/nim-waku/issues/399) purposes.
+
+JS-Waku is the second implementation of Waku v2 protocol,
+so a lot of effort on interoperability was needed.
+For example, to ensure compatibility with the nim-waku reference implementation,
+we run our [tests against wakunode2](https://github.com/status-im/js-waku/blob/90c90dea11dfd1277f530cf5d683fb92992fe141/src/lib/waku_relay/index.spec.ts#L137) as part of the CI.
+
+As we built the [web chat app](https://github.com/status-im/js-waku/tree/main/examples/web-chat),
+we were able to fine tune the API to provide a simple and succinct interface.
+You can start a node, connect to other nodes and send a message in less than ten lines of code:
+
```

Comment:
API exampe section

Reply:
There is a JS code example just below. What else would you want to include?

Reply:
at second read, that might be good enough